### PR TITLE
Adds the static 404 page to talentsearch

### LINKED ISSUE #1904: All - Create fancy 404 pages
El Jerbo (AND BROOKE) will provide some fancy design here

### Normal Mode

Screen Shot 2022-10-06 at 4 05 29 PM

### Dark Mode

Image

[Wireframe](https://www.figma.com/file/fsUTEm09c40qponRarmdGD/404-page?node-id=208%3A702)

## Acceptance Criteria
- [ ] "Report missing pages" goes to support page `/en|fr/support`. No special treatment or pre-filled things. 
- [ ] Homepage goes to homepage.
- [ ] host at /en/not-found and /fr/not-found (avoid 404)
- [ ] routing sends unknown urls to this page
- [ ] dark mode works

### PR DESCRIPTION
**Pull request by JoshBeveridge**
# ❓Purpose

Adds an error page to handle bad requests in both light and dark mode.

# 🏷️ Related issues
- resolves #1904

# 📋 TODO

- [x] Add `react-intl` strings
- [x] Add to the router
- [x] Update SVGs
- [x] Add frontend/apache redirect?

# 🧪 Testing

> **Note**
> Testing dark mode can only be done in storybook

1. Build talentsearch `npm run production --workspace=talentsearch`
2. Navigate to `/404`
3. Confirm it appears and looks like the mockup

# 📷 Screenshot

Screen Shot 2022-10-12 at 8 24 05 AM

### CODE REVIEW

File: infrastructure/php-container/src/.htaccess [hunk 1]
```diff
@@ -97,16 +104,19 @@ DirectorySlash off
...
-    RewriteCond %{REQUEST_URI} /tc-report/assets/
+    RewriteCond %{REQUEST_URI} ^/tc-report/assets/
     RewriteRule ^tc-report/assets/(.*)$ tc-report/_site/assets/$1 [L]
 
     # Send all other requests to /tc-report/_site
-    RewriteCond %{REQUEST_URI} !/tc-report/_site
+    RewriteCond %{REQUEST_URI} !^/tc-report/_site
     RewriteRule ^(.*)/?$ tc-report/_site/$1 [L]
+
+    ErrorDocument 404 /404
```
Comment (petertgiles) [inline infrastructure/php-container/src/.htaccess:119]: This really can't go live without translation, can it?

**Review by petertgiles — CHANGES_REQUESTED** [review]
This is looking so great!  A few things left to work out though.

File: frontend/talentsearch/src/js/components/404/Error404.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { INITIAL_VIEWPORTS } from "@storybook/addon-viewport";
+import { widthOf, heightOf } from "@common/helpers/storybookUtils";
+import Error404 from "./Error404";
+
+type Meta = ComponentMeta<typeof Error404>;
+type Story = ComponentStory<typeof Error404>;
+
+export default {
+  component: Error404,
+  title: "404 Page",
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/404/Error404.stories.tsx:12]: Hmm, I'm not sure what is going on here but I will take a look for sure. 👀

[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/404/Error404.stories.tsx:12]: Okay, not related to the 404 page but good catch on the bug!

I have a PR up that fixes it https://github.com/GCTC-NTGC/gc-digital-talent/pull/4281

File: infrastructure/php-container/src/.htaccess [hunk 3]
```diff
@@ -68,6 +68,13 @@ DirectorySlash off
     # Send localized talent requests also to talentsearch dist folder;
     RewriteRule ^(en|fr)/browse(/(.*))?$ frontend/talentsearch/dist/$3 [L]
 
+    # Send 404s to talentsearch dist folder (with or without public/ path prefix).
```
[the remark this answers is not in the document]
Comment (esizer) [reply infrastructure/php-container/src/.htaccess:71]: Thanks! I was not aware of this, copied over in [copy htaccess to deploy file](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245/commits/7c4d93c8be3c44616525fd454c8b874c8e03c368)

File: frontend/talentsearch/src/js/components/404/Error404.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            data-h2-display="base(flex)"
+            data-h2-gap="base(x1)"
+            data-h2-justify-content="base(center)"
+            data-h2-flex-wrap="base(wrap) p-tablet(initial)"
+          >
+            <CallToAction
+              type="link"
+              context="home"
+              content={{
+                path: "path/to/home",
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/404/Error404.tsx]: Updated in [update home path, remove stray space](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245/commits/9bca06a32d704fcd367604c1c4851c5a11b69bdd)

File: infrastructure/php-container/src/.htaccess (hunk 1 above)
Comment (esizer) [reply infrastructure/php-container/src/.htaccess:119]: Or a support page (which is the reason for https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245#discussion_r994660338 😆 

I posed the question about what to do in slack and will try to remember to bring it up during stand up

Comment (esizer) [reply infrastructure/php-container/src/.htaccess:126]: Translations completed in [add 404 French](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245/commits/5fcde8caf3d76f9652c8b607f8c8a13a15b0c986)

File: frontend/talentsearch/src/js/components/404/Error404.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                  id: "i9E0ka",
+                  description: "Link text to go to the homepage from a 404",
+                }),
+              }}
+            />
+            <CallToAction
+              type="link"
+              context="support"
+              content={{
+                path: "path/to/support",
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/404/Error404.tsx]: Added a blank support page and updated the path to it in[add blank support page](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245/commits/b2c5759fb68390846c7b76cdcd8678b127f8e625)

File: infrastructure/php-container/src/.htaccess (hunk 3 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply infrastructure/php-container/src/.htaccess:71]: Oh, careful when copying - you overwrote the original comment at the top of the file and exposed the phpinfo route in the production file.

[the remark this answers is not in the document]
Comment (petertgiles) [reply infrastructure/php-container/src/.htaccess:71]: I deployed it to DEV and it does work, FYI.

[the remark this answers is not in the document]
Comment (esizer) [reply infrastructure/php-container/src/.htaccess:71]: 😳  YIKES, thanks for catching that! [fix overwritten rules](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245/commits/dd6ab1ab950d3241e9cd4f339b11ae1fabe75498) should handle it.

**Review by petertgiles — APPROVED** [review]

**Pull request merged**